crates/iota-cluster-test/src/test_case/coin_index_test.rs

```diff
@@ -500,22 +501,16 @@ impl TestCaseImpl for CoinIndexTest {
         assert_eq!(managed_coins.len(), 40);
         assert!(managed_coins.iter().all(|c| c.balance == 5));
 
-        let mut total_coins = 0;
-        let mut cursor = None;
-        loop {
-            let page = client
+        let total_coins = GetAllPages::get_all_pages(async |cursor| {
```

Comment:
Nit: I think `GetAllPages::get_all_pages_stream` is a better candidate for this. We don't need to collect all pages, just to count the number of entries on each page.